### ISSUE TITLE
Allow configuring the default search engine

### ISSUE THREAD

**Issue by alcinnz** [alcinnz/Odysseus#79]
Or rather what I'd really like to do is to allow websurfers to register multiple search engines and to have odysseus search across all of them (or some subset matching specified tags). The [standards exist](http://www.opensearch.org/Home) for this, just no one (except [GeoNetwork opensource](https://www.geonetwork-opensource.org/)) appear to be using enough of it for this to be functional. Not even Amazon, who defined the standard. Still I'd love to implement this and to push sites to work with it.

---

So the specific experience I want to offer is to add an icon the right-hand-side of the addressbar for certain sites (namely the popular search engines and any site offering an OpenSearch descriptor) offering to "install" it. Then you'll be able to search that search engine at anytime from the addressbar just as you can with DuckDuckGo now.

Any search engines you register should if possible (i.e. for ones that support OpenSearch+RSS) be combined into a single metasearch engine builtin to Odysseus. But if not, it should be given an autocompletion entry resolving to a search via that site collectively sorted by how recently you've used it (which'll have the nice side effect that that the most recently used search engine will intelligently be the default one). That most recently used search engine will further benefit by getting to provide autocompletions of it's own.

As for reviewing the registered search engines, I'm hesitent to provide a settings dialog as a great elementary app shouldn't need one. Instead I'll tuck this into the familiar UI of the bookmarks interface under it's own tag(s).

**Comment by alcinnz** [comment]
To split this into subtasks:

- [ ] Detect OpenSearch descriptors and offer to install.
- [ ] Autoupgrade OpenSearch descriptors
- [ ] Integrate installed search engines into the autocompletion
- [ ] Track how recently a search engine is used, use that info to sort search engines.
- [ ] Integrate autocompletions from the most recently used search engine.
- [ ] Allow Prosody templates to perform web requests.
- [ ] Implement a builtin metasearch engine.
- [ ] Implement autocompletion for the metasearch engine.
- [ ] Distinguish OpenSearch+RSS support (to be used in metasearch engine) from plain OpenSearch support.
- [ ] Allow managing installed search engines through the bookmarks interface.

**Comment by alcinnz** [comment]
Depends on #75 for basic infrastructure, #78 for some of it's UI, and #82 for the UI to offer to install a search engine.
